action.yml

```
@@ -0,0 +1,45 @@
+name: Auto-tag
+description: Automatically delete and re-release tags so that other github action modules can use something similar to carets e.g. v0.1 or v3
+
+runs:
+  using: composite
+  steps:
+    - name: Validate ref
+      shell: bash
+      env:
+        REF: ${{ github.ref }}
+      run: |
+        # refs/tags/0.1.23 => 0.1.23
+        TAG=$(echo $REF | cut -c 11-)
+        if [[ "$TAG" =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+          echo "Valid semver tag $TAG"
+        else
+          echo "Invalid semver tag $TAG"
+          exit 1
+        fi
+
+    - name: Generate tag name
+      id: generate_tag_name
+      shell: bash
+      env:
+        REF: ${{ github.ref }}
+      run: |
+        # refs/tags/0.1.23 => 0.1.23
+        TAG=$(echo $REF | cut -c 11-)
+        [[ $TAG =~ ^([0-9]+)\.([0-9]+)\.([0-9]+)$ ]]
```

```suggestion
    - name: Generate tag name
      id: generate_tag_name
      shell: bash
      env:
        REF: ${{ github.ref }}
      run: |
        # refs/tags/0.1.23 => 0.1.23
        TAG=$(echo $REF | cut -c 11-)
        if ! [[ "$TAG" =~ ^([0-9]+)\.([0-9]+)\.([0-9]+)$ ]]; then
          echo "Invalid semver tag $TAG"
          exit 1
        fi
        echo "Valid semver tag $TAG"
```

I don't think it's useful to have the validation step separate - it just results in code duplication.

---

Have combined steps